web/public/js/init-cdf.js

```diff
@@ -0,0 +1,302 @@
+(function() {
+  'use strict';
+
+
+  var resolveDojo = function($q, $stateParams, cdDojoService) {
+    var dfd = $q.defer();
+    if($stateParams.legacyId && _.isNumber(parseInt($stateParams.legacyId))) {
+      cdDojoService.list({mysqlDojoId: parseInt($stateParams.legacyId)}, function (data) {
+        dfd.resolve(data[0]);
+      }, function (err) {
+        dfd.reject(err);
+      });
+    } else if( ($stateParams.dojoId && !_.isNumber($stateParams.dojoId) ) ||
+      ($stateParams.id && !_.isNumber($stateParams.dojoId) ) ) {
+        var id = $stateParams.dojoId || $stateParams.id;
+        cdDojoService.load(id,
+        function (data) {
+          dfd.resolve(data);
+        }, function (err) {
+          dfd.reject(err);
+      });
+    } else if($stateParams.country && $stateParams.path && _.isString($stateParams.country) && _.isString($stateParams.path)) {
+      cdDojoService.find({
+        urlSlug: $stateParams.country + '/' + $stateParams.path
+      }, function(data) {
+        dfd.resolve(data);
+      }, function(err) {
+        dfd.reject(err);
+      });
+    } else {
+      dfd.reject(new Error('No Dojo found.'));
+    }
+    return dfd.promise;
+  };
+
+  var failCb = function(err){
+    return {err: err};
+  };
+
+  var winCb = function(data){
+    return {data: data};
+  };
+
+  var resolves = {
```

Comment:
We should probably centralise all our common resolves between init-master and init-cdf.